src/map/entities/battleentity.cpp

```diff
@@ -733,7 +769,7 @@ void CBattleEntity::SetMLevel(uint8 mlvl)
 
 void CBattleEntity::SetSLevel(uint8 slvl)
 {
-    m_slvl = (slvl > (m_mlvl >> 1) ? (m_mlvl == 1 ? 1 : (m_mlvl >> 1)) : slvl);
+    m_slvl = (slvl > (m_mlvl * map_config.sj_level_ratio) ? (m_mlvl == 1 ? 1 : (m_mlvl * map_config.sj_level_ratio)) : slvl);
```

Comment:
Thanks. I'll have to do some more testing on this one, unless you're going to do your patch instead.